beacon_node/beacon_chain/src/attestation_rewards.rs

```diff
@@ -0,0 +1,185 @@
+use beacon_chain::{BeaconChain, BeaconChainTypes};
+use eth2::lighthouse::attestation_rewards::{IdealAttestationRewards, TotalAttestationRewards};
+use eth2::lighthouse::StandardAttestationRewards;
+use participation_cache::ParticipationCache;
+use safe_arith::SafeArith;
+use slog::{debug, Logger};
+use state_processing::{
+    common::altair::BaseRewardPerIncrement,
+    per_epoch_processing::altair::{participation_cache, rewards_and_penalties::get_flag_weight},
+};
+use std::{collections::HashMap, sync::Arc};
+use store::consts::altair::PARTICIPATION_FLAG_WEIGHTS;
+use types::consts::altair::WEIGHT_DENOMINATOR;
+
+use types::{Epoch, EthSpec};
+
+use crate::{beacon_chain, BeaconChainError};
+
+impl<T: BeaconChainTypes> BeaconChain<T> {
+    pub fn compute_attestation_rewards(
+        &self,
+        epoch: Epoch,
+        chain: Arc<BeaconChain<T>>,
+        mut validators: Vec<usize>,
+        log: Logger,
+    ) -> Result<StandardAttestationRewards, BeaconChainError> {
+        debug!(log, "computing attestation rewards"; "epoch" => epoch, "validator_count" => validators.len());
+
+        //--- Get state ---//
+        let spec = &chain.spec;
+
+        let execution_optimistic = chain.is_optimistic_or_invalid_head()?;
+
+        let state_slot = (epoch + 1).end_slot(T::EthSpec::slots_per_epoch());
+
+        let state_root = chain.state_root_at_slot(state_slot)?;
+
+        let state = chain.get_state(&state_root.unwrap(), Some(state_slot))?;
```

Comment:
If we can't get the `state_root` or the `state` (they are `None`) we should immediately return an error. It'll save a lot of error handling later

For the state something like:

```rust
.ok_or(BeaconChainError::MissingBeaconState(state_root))?;
```